Update uninstall methods for single user (#8675)
NixOS/nix: doc/manual/src/installation/uninstall.md

# Uninstalling Nix

## Sections
- [Single User](#single-user)
  - [Manual Method](#manual-method)
    - [Step 1: Restore Modified Files (if applicable)](#step-2-restore-modified-files-if-applicable)
    - [Step 2: Delete Nix Files](#step-4-delete-nix-files)
- [Multi User](#multi-user)
    - [Linux](#linux)
    - [macOS](#macos)

## Single User

> ### If you have a [single-user installation](./installing-binary.md#single-user-installation) of Nix
> ### Manual Method

Reviewer: ```suggestion

These instructions apply for a [single-user installation](./installing-binary.md#single-user-installation) of Nix.
```

> #### Step 1: Restore Modified Files (if applicable)

Reviewer: ```suggestion
### Step 1: Restore Modified Files
```

If they don't exist it's a no-op anyway

> 1. Nix may have modified certain files on your system. If you have created backup copies of these files before installing Nix, you can restore them. Here are the instructions for bash and zsh files:

Reviewer: It's not perfectly correct to call them configuration files. Maybe @roberth has a better suggestion.

```suggestion
Nix will create backup copies of shell configuration files before installation, an you can restore them.
Here are the instructions for Bash and Zsh:
```

There is no second item in that list, so don't make it a list. Use one line per sentence, it's easier to review and make focused suggestions. This is the general rule here, even if it's not consistently implemented yet. Use the project names with their original spelling (Bash, Zsh). Use monospace for command names (`bash`, `zsh`, but that's not applicable here). I will add a contribution guide for the manual this year, promised.

>    - Bash: Nix may have modified the `/etc/bash.bashrc` or `~/.bash_profile` file. If you have created a backup of this file before installing Nix, you can restore it using the following commands:

Reviewer: Dedent. I don't think Nix *may* have done it, pretty sure it just did. I doubt this introductory blurb is even needed.

     ```bash

>      sudo cp /etc/bash.bashrc.backup-before-nix /etc/bash.bashrc
>      sudo cp ~/.bash_profile.backup-before-nix ~/.bash_profile

Reviewer: Why not

```suggestion
     sudo mv /etc/bash.bashrc.backup-before-nix /etc/bash.bashrc
     sudo mv ~/.bash_profile.backup-before-nix ~/.bash_profile
```

Because if those files exist we're done; otherwise error showing they don't

     ```

>      Note: After restoring the file, you may need to close and reopen any bash terminal sessions to ensure they are using the restored configurations.

Reviewer: You sure have to. Same for the other note.

```suggestion
     > **Note**
     >
     > After restoring the file, you have to close and reopen any Bash sessions to ensure they are using the restored configurations.
```

We use this type of highlighting for admonitions everywhere

   - Zsh: Nix may have modified the `~/.zshrc` file. If you have created a backup of this file before installing Nix, you can restore it using the following command:

     ```bash
     sudo cp ~/.zshrc.backup-before-nix ~/.zshrc
     ```

     Note: After restoring the file, you may need to close and reopen any zsh terminal sessions to ensure they are using the restored configurations.

> #### Step 2: Delete Nix Files

Reviewer: ```suggestion
### Step 2: Delete Nix Files
```

> 1. Run the following command in a terminal to delete the files that Nix added to your system:

Reviewer: This doesn't have to be a list, dedent.

   ```bash
   sudo rm -rf "/etc/nix" "$HOME/.nix-profile" "$HOME/.nix-defexpr" "$HOME/.nix-channels" "$HOME/.cache/nix" "$NIX_ROOT" "/nix"
   ```

> You have successfully uninstalled Nix from your system. Remember to double-check the commands before executing them and ensure that you have the necessary permissions to perform the uninstallation steps.

Reviewer: That's a bit too much promised. The reminder should be a warning admonition *before* the commands.

> If you ever need to use Nix again in the future, you can reinstall it following the installation instructions provided by the Nix documentation.

Reviewer: Link to those.

## Single User

### Alternative:
 #### Step 1: 

>  Run the script located at `scripts/uninstall-single-user.sh`

Reviewer: Imagine you're reading this as a first time user that got in trouble with installation or just wants to bail out. Where is that location? What exactly would you have to do to run that script? Currently there is no indication at all, and you "just have to know" that you mean something in the GitHub repository. I insist we avoid this "just have to know" situation at all cost. It got Nix the reputation of being badly documented and inscrutable.

Usually at this point I'd suggest to make it a link to the repo, but that script must match the current Nix version to be sure it will do the right thing. Also it would be quite a few additional steps to run it off the repo. What about we include it as a command in the binary? We could use that opportunity to take inventory of all the file Nix touches, in one central location.

Author: That's a really good point, thanks! I definitely agree with you, and I'd like to help Nix be more consistent; swapping this to part of the binary makes sense and I can get to work on that.

Author: @fricklerhandwerk Where should I put it in the dir structure? I don't typically develop in C++ but I'm learning as I go 😄

Reviewer: We'll discuss among maintainers how to proceed here in general.
